src/components/DataViewer/DataViewer.tsx

```diff
@@ -84,31 +70,23 @@ interface DataViewerProps {
     header: ReactElement | string;
     isOpen: boolean;
     onRequestClose: () => void;
+    ariaLabelledBy: string;
```

Comment:
did you say aria label is optional? doesn't look it here.
(can we set it by default to something if you want it optional, or just make it non-optional and force devs to put a label? happy with either)

Reply:
Because the modal has an aria role of `dialog` an `aria-labelledby` property (different to `aria-label`) has to be passed to the `Modal` component. If this is not setup on the modal, then accessibility standards are violated - that's why it hasn't been set as optional